src/elements/src/index.ts

```diff
@@ -0,0 +1,55 @@
+import { LifeCycles } from 'single-spa';
+import { NgElement } from '@angular/elements';
+
+import {
+  SingleSpaAngularElementsOptions,
+  BootstrappedSingleSpaAngularElementsOptions,
+} from './types';
+
+const defaultOptions: BootstrappedSingleSpaAngularElementsOptions = {
+  element: null!,
+  ngModuleRef: null,
+  bootstrapFunction: null!,
+  withProperties: undefined,
+  domElementGetter: undefined,
+};
+
+async function bootstrap(options: BootstrappedSingleSpaAngularElementsOptions) {
+  if (options.ngModuleRef !== null) {
+    return;
+  }
+
+  options.ngModuleRef = await options.bootstrapFunction();
```

Comment:
How do we tell angular to mount to a specific dom element container? The domElementGetter should provide that container.

Also, why are we calling bootstrapFunction inside of single-spa's `bootstrap` lifecycle instead of `mount` lifecycle? Doesn't calling `bootstrapFunction` mount the angular application to a dom element? If, so we definitely need to move that to the `mount` function.

Reply:
AFAIK the `bootstrap` lifecycle is called once and `mount` is called every time the app is rendered. Modules that expose custom elements should be bootstrapped once.

Reply:
👍 sounds good